x/participationrewards/keeper/submodule_umee.go

```diff
@@ -0,0 +1,210 @@
+			"store/leverage/key",
+			umeetypes.KeyUTokenSupply(supply.Denom),
+			sdk.NewInt(-1),
+			types.ModuleName,
+			UmeeUTokenSupplyUpdateCallbackID,
+			0,
+		) // query utoken supply
+
+		return false
+	})
+
+	// TODO: check module spendable coins retrieval
+	// assuming that module account is not a vesting account so there
+	// will be no locked coins to subtract from the total balance
+	k.IteratePrefixedProtocolDatas(ctx, types.GetPrefixProtocolDataKey(types.ProtocolDataTypeUmeeLeverageModuleBalance), func(idx int64, _ []byte, data types.ProtocolData) bool {
+		ibalance, err := types.UnmarshalProtocolData(types.ProtocolDataTypeUmeeLeverageModuleBalance, data.Data)
+		if err != nil {
+			return false
+		}
+		balance, _ := ibalance.(*types.UmeeLeverageModuleBalanceProtocolData)
+		accountPrefix := banktypes.CreateAccountBalancesPrefix(authtypes.NewModuleAddress(umeetypes.LeverageModuleName))
+
+		// update leverage module balance
+		k.IcqKeeper.MakeRequest(
+			ctx,
+			connectionData.ConnectionID,
+			connectionData.ChainID,
+			icstypes.BankStoreKey,
+			append(accountPrefix, []byte(balance.Denom)...),
+			sdk.NewInt(-1),
+			types.ModuleName,
+			UmeeLeverageModuleBalanceUpdateCallbackID,
+			0,
+		) // query leverage module balance
+
+		return false
+	})
+	// umee-types total borrowed update
+	k.IteratePrefixedProtocolDatas(ctx, types.GetPrefixProtocolDataKey(types.ProtocolDataTypeUmeeTotalBorrows), func(idx int64, _ []byte, data types.ProtocolData) bool {
+		iborrows, err := types.UnmarshalProtocolData(types.ProtocolDataTypeUmeeTotalBorrows, data.Data)
+		if err != nil {
+			return false
+		}
+		borrows, _ := iborrows.(*types.UmeeTotalBorrowsProtocolData)
+
+		// update total borrows for a denom
+		k.IcqKeeper.MakeRequest(
+			ctx,
+			connectionData.ConnectionID,
+			connectionData.ChainID,
+			"store/leverage/key",
+			umeetypes.KeyAdjustedTotalBorrow(borrows.Denom),
+			sdk.NewInt(-1),
+			types.ModuleName,
+			UmeeTotalBorrowsUpdateCallbackID,
+			0,
+		) // query leverage module balance
+
+		return false
+	})
+}
+
+func getDenomFromProof(proof *cmtypes.Proof, addr []byte) (string, error) {
+	denom, err := utils.DenomFromRequestKey(proof.Key, addr)
+	if err != nil {
+		return "", err
+	}
+	if proof.ProofType == types.ProofTypeLeverage {
+		denom = denom[:len(denom)-1]
+	}
+	return denom, err
+}
+
+func (u UmeeModule) ValidateClaim(ctx sdk.Context, k *Keeper, msg *types.MsgSubmitClaim) (uint64, error) {
+	amount := uint64(0)
+	_, addr, err := bech32.DecodeAndConvert(msg.UserAddress)
+
+	for _, proof := range msg.Proofs {
+		// determine denoms from keys
+		if proof.Data == nil {
+			continue
+		}
+
+		udenom, err := getDenomFromProof(proof, addr)
+		if err != nil {
+			return 0, err
+		}
+		uToken, err := bankkeeper.UnmarshalBalanceCompat(k.cdc, proof.Data, udenom)
+		if err != nil {
+			return 0, err
+		}
+		token, err := umee.ExchangeUToken(ctx, uToken, k)
+		if err != nil {
+			return 0, err
+		}
+		amount += token.Amount.Uint64()
```

Comment:
I just noticed we aren't checking whether the denom we have here is an allowed value. 

Xcclookup does this check already, but that doesn't stop a user creating a message containing a u/ denom we don't support, which will currently get through this validation.

Check out submodule_liquid. It looks up the remote denom in the allowed list to validate it is one we support.